src/main/java/net/fabricmc/loader/FabricMappingResolver.java

```diff
@@ -36,44 +38,52 @@
 		private final Map<EntryTriple, String> methodNames = new HashMap<>();
 	}
 
-	FabricMappingResolver(Supplier<Mappings> mappingsSupplier, String targetNamespace) {
+	FabricMappingResolver(Supplier<TinyTree> mappingsSupplier, String targetNamespace) {
 		this.mappingsSupplier = mappingsSupplier;
 		this.targetNamespace = targetNamespace;
-		namespaces = Collections.unmodifiableSet(new HashSet<>(mappingsSupplier.get().getNamespaces()));
+		namespaces = Collections.unmodifiableSet(new HashSet<>(mappingsSupplier.get().getMetadata().getNamespaces()));
 	}
 
 	protected final NamespaceData getNamespaceData(String namespace) {
-		return namespaceDataMap.computeIfAbsent(namespace, (ns) -> {
+		return namespaceDataMap.computeIfAbsent(namespace, (fromNamespace) -> {
 			if (!namespaces.contains(namespace)) {
 				throw new IllegalArgumentException("Unknown namespace: " + namespace);
 			}
 
 			NamespaceData data = new NamespaceData();
-			Mappings mappings = mappingsSupplier.get();
+			TinyTree mappings = mappingsSupplier.get();
 			Map<String, String> classNameMap = new HashMap<>();
-			Function<String, String> classMapper = (s) -> classNameMap.computeIfAbsent(s, (cname) -> cname.replace('/', '.'));
-			Function<EntryTriple, EntryTriple> tripletMapper = (s) -> new EntryTriple(classMapper.apply(s.getOwner()), s.getName(), s.getDesc());
 
-			for (ClassEntry classEntry : mappings.getClassEntries()) {
-				String fromClass = classMapper.apply(classEntry.get(ns));
-				String toClass = classMapper.apply(classEntry.get(targetNamespace));
+			for (ClassDef classEntry : mappings.getClasses()) {
+				String fromClass = computeIfAbsentClassName(classNameMap, classEntry.getName(fromNamespace));
+				String toClass = computeIfAbsentClassName(classNameMap, classEntry.getName(targetNamespace));
 
 				data.classNames.put(fromClass, toClass);
 				data.classNamesInverse.put(toClass, fromClass);
-			}
 
-			for (FieldEntry fieldEntry : mappings.getFieldEntries()) {
-				data.fieldNames.put(tripletMapper.apply(fieldEntry.get(ns)), fieldEntry.get(targetNamespace).getName());
-			}
+				String mappedClassName = computeIfAbsentClassName(classNameMap, fromClass);
 
-			for (MethodEntry methodEntry : mappings.getMethodEntries()) {
-				data.methodNames.put(tripletMapper.apply(methodEntry.get(ns)), methodEntry.get(targetNamespace).getName());
+				putDescriptored(fromNamespace, classEntry.getFields(), data.fieldNames, mappedClassName);
+				putDescriptored(fromNamespace, classEntry.getMethods(), data.methodNames, mappedClassName);
 			}
 
 			return data;
 		});
 	}
 
+	private String computeIfAbsentClassName(Map<String, String> classNameMap, String s) {
+		return classNameMap.computeIfAbsent(s, (cname) -> cname.replace('/', '.'));
+	}
+
+	private <T extends Descriptored> void putDescriptored(String fromNamespace, Collection<T> descriptoredList,
+														  Map<EntryTriple, String> putInto, String fromClass) {
```

Comment:
Please remove the weird line break and call it recordMember or similar

Reply:
it just looks weird in Github

Reply:
up to 160 columns are fine imo, if you continue lines please use 2 more tabs than the original indent